packages/hoprd/src/index.ts

```diff
@@ -73,196 +49,45 @@ const metric_latency = create_histogram_with_buckets(
-      'Allow connections to other nodes running on private addresses [env: HOPRD_ALLOW_PRIVATE_NODE_CONNECTIONS]',
-    default: false
-  })
-  .option('disableApiAuthentication', {
-    boolean: true,
-    describe: 'Disable authentication for the API endpoints [env: HOPRD_DISABLE_API_AUTHENTICATION]',
-    default: false
-  })
-  .option('testAnnounceLocalAddresses', {
-    hidden: true,
-    boolean: true,
-    describe: 'For testing local testnets. Announce local addresses [env: HOPRD_TEST_ANNOUNCE_LOCAL_ADDRESSES]',
-    default: false
-  })
-  .option('testPreferLocalAddresses', {
-    hidden: true,
-    boolean: true,
-    describe: 'For testing local testnets. Prefer local peers to remote [env: HOPRD_TEST_PREFER_LOCAL_ADDRESSES]',
-    default: false
-  })
-  .option('testUseWeakCrypto', {
-    hidden: true,
-    boolean: true,
-    describe: 'weaker crypto for faster node startup [env: HOPRD_TEST_USE_WEAK_CRYPTO]',
-    default: false
-  })
-  .option('testNoDirectConnections', {
-    hidden: true,
-    boolean: true,
-    describe:
-      'NAT traversal testing: prevent nodes from establishing direct TCP connections [env: HOPRD_TEST_NO_DIRECT_CONNECTIONS]',
-    default: false
-  })
-  .option('testNoWebRTCUpgrade', {
-    hidden: true,
-    boolean: true,
-    describe:
-      'NAT traversal testing: prevent nodes from establishing direct TCP connections [env: HOPRD_TEST_NO_WEB_RTC_UPGRADE]',
-    default: false
-  })
-  .option('testNoUPNP', {
-    hidden: true,
-    boolean: true,
-    describe:
-      'NAT traversal testing: disable automatic detection of external IP address using UPNP [env: HOPRD_TEST_NO_UPNP]',
-    default: false
-  })
-  .option('heartbeatInterval', {
-    number: true,
-    describe:
-      'Interval in milliseconds in which the availability of other nodes get measured [env: HOPRD_HEARTBEAT_INTERVAL]',
-    default: HEARTBEAT_INTERVAL
-  })
-  .option('heartbeatThreshold', {
-    number: true,
-    describe:
-      "Timeframe in milliseconds after which a heartbeat to another peer is performed, if it hasn't been seen since [env: HOPRD_HEARTBEAT_THRESHOLD]",
-    default: HEARTBEAT_THRESHOLD
-  })
-  .option('heartbeatVariance', {
-    number: true,
-    describe: 'Upper bound for variance applied to heartbeat interval in milliseconds [env: HOPRD_HEARTBEAT_VARIANCE]',
-    default: HEARTBEAT_INTERVAL_VARIANCE
-  })
-  .option('networkQualityThreshold', {
-    number: true,
-    describe: 'Miniumum quality of a peer connection to be considered usable [env: HOPRD_NETWORK_QUALITY_THRESHOLD]',
-    default: NETWORK_QUALITY_THRESHOLD
-  })
-  .option('onChainConfirmations', {
-    number: true,
-    describe: 'Number of confirmations required for on-chain transactions [env: HOPRD_ON_CHAIN_CONFIRMATIONS]',
-    default: CONFIRMATIONS
-  })
-  .showHidden('show-hidden', 'show all options, including debug options')
-  .strict()
-  .wrap(Math.min(120, yargsInstance.terminalWidth()))
-  .parseSync()
+type CliArgs = {
```

Comment:
Not sure if it would be possible to define this type in Rust and just use the wasm-bindgen generated TS type here?

Reply:
changed